clients/clearly-defined/src/main/kotlin/ClearlyDefinedService.kt

```diff
@@ -268,3 +274,13 @@ interface ClearlyDefinedService {
         @Path("toolVersion") toolVersion: String
     ): ResponseBody
 }
+
+suspend fun ClearlyDefinedService.getCurationsChunked(
+    coordinates: Collection<Coordinates>,
+    chunkSize: Int = ClearlyDefinedService.MAX_REQUEST_CHUNK_SIZE
+): Map<Coordinates, Curation> =
+    buildMap {
+        coordinates.chunked(chunkSize).forEach { chunk ->
+            getCurations(chunk).values.forEach { putAll(it.curations) }
+        }
+    }
```

Comment:
Would it make sense to run the requests in parallel using a `map { async { ... }}.awaitAll()` construct?

Reply:
I was asking myself the same thing as well but was undecided, so I wanted to wait whether it would come up during the review. As it does, I'll change it 😸

Reply:
But wait, now I remember, it basically boils down to the same question: Should the function be blocking. Because `async` needs a `CoroutineScope`, so I'd have to wrap it in `runBlocking` again.

Reply:
I've reworked the code quite a bit, please have another look.

Reply:
You don't need `runBlocking` for that, you could have wrapped the async code with `coroutineScope {}` instead and kept it a suspend function.

Reply:
Ah, right. Well, now the extension functions are higher level convenience functions that already do the blocking, which was anyway required by all current callers. So I believe it's ok that way, but should we ever need to change it again, we should do like you propose.